File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0, r = 1 , maxP =0;

        while(r < prices.length){
            if(prices[r] > prices[l]){
                int currP = prices[r] - prices[l];
                maxP = Math.max(currP,maxP);
            } else {
                l = r;
            }
            r++;
        }
        return maxP;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfit = 0
        for i in range(1, len(prices) - 1):
            if prices[i] - minPrice > maxProfit:
                maxProfit = prices[i] - minPrice
            elif prices[i] < minPrice:
                minPrice = min(prices[i],minPrice)
        return maxProfit
        